### DISCUSSION TITLE
Prevent cloning the main repository

### DISCUSSION THREAD

**Discussion by ibeerens** [community/community#53758] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have for example a main repository. Under the main repository, there will be sub-repositories. The structure looks like this:

customers
  customer-a
    code
  customer-b
    code
  customer-c
    code
 
Is it possible to prevent cloning the main repository (customers)? I want only allowing to clone a sub-repository such as customer-a for example.

**Comment by airtower-luna** [comment]
What exactly do you mean by "sub-repositories"? That's not a Git term.

If you mean directories within the repository, what you want is not possible. Access to a repository is all-or-nothing, you cannot give someone access to only part of a repository.

If you mean [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules): A submodule is just a pointer at another repository, so you'll need to adjust access rights of the referenced repositories. And preferably just point your customers at their individual repositories.

**Comment by ibeerens** [comment]
Yes I mean directories in a repository. If we make a repository for every customer the structure will be very big. Are there ways to structure the repositories?

**Comment by airtower-luna** [comment, marked as answer]
Not really, though I've seen people request it before:

* #17662

**Comment by ibeerens** [comment]
Thanks for the answer.
